yt_dlp/extractor/wimtv.py

```diff
@@ -0,0 +1,163 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    parse_duration,
+    urlencode_postdata,
+    ExtractorError,
+)
+
+
+class WimTVIE(InfoExtractor):
+    _player = None
+    _UUID_RE = r'[\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12}'
+    _VALID_URL = r'''(?x)
+        https?://platform.wim.tv/
+        (?:
+            (?:embed/)?\?
+            |\#/webtv/.+?/
+        )
+        (?P<type>vod|live|cast)[=/]
+        (?P<id>%s).*?''' % _UUID_RE
+    _TESTS = [{
+        # vod stream
+        'url': 'https://platform.wim.tv/embed/?vod=db29fb32-bade-47b6-a3a6-cb69fe80267a',
+        'md5': 'db29fb32-bade-47b6-a3a6-cb69fe80267a',
+        'info_dict': {
+            'id': 'db29fb32-bade-47b6-a3a6-cb69fe80267a',
+            'ext': 'm3u8',
```

Comment:
This should be `mp4`. Otherwise user will end up with an 'm3u8' file which is not a proper video file, but a playlist file instead